promql/testdata/selectors.test

```diff
@@ -20,6 +20,19 @@ eval instant at 8000s rate(http_requests{group=~"pro.*"}[1m])
 	{job="api-server", instance="0", group="production"} 1
 	{job="api-server", instance="1", group="production"} 2
 
+eval instant at 8000s rate(http_requests{group=~"pro.*"}[1m] @ 8000)
+	{job="api-server", instance="0", group="production"} 1
+	{job="api-server", instance="1", group="production"} 2
+
+# TODO(codesome): Why does this not work?
+#eval instant at 1s rate(http_requests{group=~"pro.*"}[1m] @ 8000)
+#	{job="api-server", instance="0", group="production"} 1
+#	{job="api-server", instance="1", group="production"} 2
```

Comment:
@brian-brazil I am confused here about the PromQL execution. Here the matrix selector is apparently not getting any samples from TSDB though the query hints are correct. Also, it work from eval time 7020s onwards and not before (if that gives you some hint). 

This is the same case with some other places where the selector returns no result after expanding. (One place was when it crossed the 5m staleness. But since with @ timestamp the query bounds changes and within that 5m, I expected it to work.)

Any idea what might be happening?

Reply:
https://github.com/prometheus/prometheus/pull/8121#discussion_r526827566

Keep in mind that the TSDB doesn't use those hints.

Reply:
> Keep in mind that the TSDB doesn't use those hints.

:facepalm: This alone fixed all remaining of my PromQL test failures. I missed that we get the querier for some time range. After setting the mint and maxt properly, it all works fine now. Pushing shortly.